src/rpcmining.cpp

```diff
@@ -211,7 +211,7 @@ UniValue setgenerate(const UniValue& params, bool fHelp)
     if (params.size() > 0)
         fGenerate = params[0].get_bool();
 
-    int nGenProcLimit = -1;
+    int nGenProcLimit = GetArg("-genproclimit", DEFAULT_GENERATE_THREADS);
```

Comment:
Making `getmininginfo` push back the "correct" value `nGenProcLimit` may involve some more work. But at least this PR did not introduce the issue. The PR is not perfect, but improves things considerably.